### ISSUE TITLE
Conditional expression is not matched by deep expression

### ISSUE THREAD

**Issue by mjambon** [semgrep/semgrep#3953]
**Describe the bug**

**To Reproduce**

All 3 methods should match but currently the first one doesn't.

pattern:
```
<... a ...>;
...;
<... b ...>
```

```
public class FooClass{
    public bool Bad1(){
        a;
        if(b){}
    }
    public bool Bad2(){
        a;
        cond = b;
        if(cond){}
    }
    public bool Bad3(){
        a;
        if(true){
            b;
        }
    }
}
```

https://semgrep.dev/s/mjambon:match-conditional

- [ ] P0: blocking your adoption of Semgrep or workflow
- [x] P1: important to fix or quite annoying
- [ ] P2: regular bug that should get fixed

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-417/conditional-expression-is-not-matched-by-deep-expression.

**Comment by nbrahms** [comment]
Fixed in `cf73f8af36301ac7aa362699652608c66eb2a295`
